pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -46,14 +46,73 @@ var (
 	}
 )
 
-// ApplyParameters applies the params from a TaskRun.Input.Parameters to a TaskSpec
-func ApplyParameters(ctx context.Context, spec *v1.TaskSpec, tr *v1.TaskRun, defaults ...v1.ParamSpec) *v1.TaskSpec {
+// applyStepActionParameters applies the params from a TaskRun.Input.Parameters to a TaskSpec
+func applyStepActionParameters(ctx context.Context, step *v1.Step, spec *v1.TaskSpec, tr *v1.TaskRun, stepParams v1.Params, defaults []v1.ParamSpec) *v1.Step {
+	if stepParams != nil {
+		stringR, arrayR, objectR := getTaskParameters(ctx, spec, tr, spec.Params...)
+		stepParams = stepParams.ReplaceVariables(stringR, arrayR, objectR)
+	}
+	stringReplacements := map[string]string{}
+	arrayReplacements := map[string][]string{}
+	objectReplacements := map[string]map[string]string{}
```

Comment:
is objectReplacements used in this function? Looks like it is created, updated but not used in container.ApplyStepReplacements

Reply:
Yes, good catch. I cleaned up the usage but forgot this.

Reply:
fixed